src/celeritas/optical/OpticalCollector.cc

```diff
@@ -152,21 +149,49 @@ AuxId OpticalCollector::optical_aux_id() const
  */
 OpticalAccumStats OpticalCollector::exchange_counters(AuxStateVec& aux) const
 {
-    auto& state = dynamic_cast<OpticalOffloadStateBase&>(
-        aux.at(this->offload_aux_id()));
-    return std::exchange(state.accum, {});
+    auto& state = dynamic_cast<optical::CoreStateBase&>(
+        aux.at(this->optical_aux_id()));
+    auto& accum = state.accum();
+
+    if (auto id = this->cherenkov_aux_id())
+    {
+        auto& gen = dynamic_cast<GeneratorStateBase const&>(aux.at(id));
+        accum.cherenkov = gen.accum;
+    }
+    if (auto id = this->scintillation_aux_id())
+    {
+        auto& gen = dynamic_cast<GeneratorStateBase const&>(aux.at(id));
+        accum.scintillation = gen.accum;
+    }
+
+    return std::exchange(accum, {});
 }
 
 //---------------------------------------------------------------------------//
 /*!
  * Get info on the number of tracks in the buffer.
  */
 auto OpticalCollector::buffer_counts(AuxStateVec const& aux) const
-    -> OpticalBufferSize const&
+    -> OpticalBufferSize
 {
-    auto& state = dynamic_cast<OpticalOffloadStateBase const&>(
-        aux.at(this->offload_aux_id()));
-    return state.buffer_size;
+    OpticalBufferSize result;
+
+    auto const& state = dynamic_cast<optical::CoreStateBase const&>(
+        aux.at(this->optical_aux_id()));
+    result.photons = state.counters().num_pending;
+
+    if (auto id = this->cherenkov_aux_id())
+    {
+        auto& gen = dynamic_cast<GeneratorStateBase const&>(aux.at(id));
+        result.distributions += gen.buffer_size;
+    }
+    if (auto id = this->scintillation_aux_id())
+    {
+        auto& gen = dynamic_cast<GeneratorStateBase const&>(aux.at(id));
+        result.distributions += gen.buffer_size;
+    }
```

Comment:
Could you write this as:
```suggestion
    if (auto* gen = dynamic_cast<GeneratorStateBase const*>(&aux.at(id)))
    {
        result.distributions += gen->buffer_size;
    }
```

Reply:
Hmm maybe not since the IDs need to be valid?